app/models/manageiq/providers/nsxt/manager_mixin.rb

```diff
@@ -22,14 +22,109 @@ def translate_exception(err)
+                        :label => _("SSL without validation"),
+                        :value => "ssl-no-validation"
+                      },
+                      {
+                        :label => _("SSL"),
+                        :value => "ssl-with-validation"
+                      },
+                      {
+                        :label => _("Non-SSL"),
+                        :value => "non-ssl"
+                      }
+                    ]
+                  },
+                  {
+                    :component  => "text-field",
+                    :id         => "endpoints.default.hostname",
+                    :name       => "endpoints.default.hostname",
+                    :label      => _("Hostname (or IPv4 or IPv6 address)"),
+                    :isRequired => true,
+                    :validate   => [{:type => "required-validator"}]
+                  },
+                  {
+                    :component    => "text-field",
+                    :id           => "endpoints.default.port",
+                    :name         => "endpoints.default.port",
+                    :label        => _("API Port"),
+                    :type         => "number",
+                    :isRequired   => true,
+                    :validate     => [{:type => "required-validator"}],
+                    :initialValue => 443,
+                  },
+                  {
+                    :component  => "text-field",
+                    :id         => "authentications.default.userid",
+                    :name       => "authentications.default.userid",
+                    :label      => _("Username"),
+                    :isRequired => true,
+                    :validate   => [{:type => "required-validator"}],
+                  },
+                  {
+                    :component  => "password-field",
+                    :id         => "authentications.default.password",
+                    :name       => "authentications.default.password",
+                    :label      => _("Password"),
+                    :type       => "password",
+                    :isRequired => true,
+                    :validate   => [{:type => "required-validator"}],
+                  },
+                ],
+              },
+            ],
+          },
+        ]
+      }.freeze
+    end
+
+    def verify_credentials(args)
+      endpoint = args.dig("endpoints", 'default')
+      hostname, security_protocol, port = endpoint&.values_at('hostname', 'security_protocol', 'port')
+      authentication = args.dig("authentications", "default")
+      userid, password = authentication&.values_at('userid', 'password')
+      !raw_connect(base_url(security_protocol, hostname, port), userid, password, security_protocol == 'ssl-with-validation')
+    end
   end
 
   def description
     "Vmware NSX-T"
   end
 
   def base_url(protocol, server, port)
-    scheme = %w[ssl ssl-with-validation].include?(protocol) ? "https" : "http"
```

Comment:
Not sure if we need a migration to update these values in the DB, I guess no one actually created this provider yet. Can you please verify @agrare?

Reply:
I think we at KPN are the only users of this providers. And if necessary we can change it by hand.

Reply:
The NSX-T provider will be introduced in the ManageIQ Kasparov release, so no database migration is needed.